bundles/org.eclipse.swt/Eclipse SWT/gtk/org/eclipse/swt/widgets/Control.java

```diff
@@ -5888,6 +5890,10 @@ public boolean setParent (Composite parent) {
 	this.parent = parent;
 	setZOrder (null, false, true);
 	reskin (SWT.ALL);
+	// restore focus to the last Control that had it, if focus is now gone
+	if (focusControlBeforeReparent != null && display.getFocusControl() == null) {
+		focusControlBeforeReparent.setFocus();
```

Comment:
Can `focusControlBeforeReparent` be disposed as a result of reparenting? Maybe better to check that.

Beside this, works as expected.